### PR TITLE
Contribute to app-i18n/fcitx-bamboo

### PR DESCRIPTION
**Pull request by vuongtuha**

### CODE REVIEW

**Comment by Universebenzene** [comment]
@bekcpear 我先解释一下，这位越南网友是想给gentoo-zh提交fcitx5越南输入法的pr的，我去帮他review修ebuild，然后蔡博士骂我是跪舔南洋人的舔狗，我觉得再在那交流实在自取其辱于是就推荐他到这来pr了，希望iYI菊苣理解下

另外这位网友我拉他进老牌群了，iYI菊苣回头清理用户的时候注意下

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: This should be the default behavior of `cmake.eclass`. So you don't need to write the `src_configure` phase out

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:31]: This should be the default behavior of c`xdg.eclass`. You don't need to write the `pkg_postinst` phase out

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: This is not a live ebuild, and here has no live ebuild, so these lines should be removed.

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: Should be replaced with a single `SRC_URI=".."` line due to this is not a live ebuild.

And, add a `-> ${P}.tar.xz` to the `SRC_URI` to make the distfile more clear.

```
SRC_URI="https://download.fcitx-im.org/fcitx5/${MY_PN}/${MY_P}.tar.xz -> ${P}.tar.xz"
```

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: Should be `>=dev-lang/go-1.18` because of the minimal go version of bamboo-core:
https://github.com/BambooEngine/bamboo-core/blob/f0ca534819fa84ee37df9277ff2449d82ed1ec3e/go.mod#L3
https://github.com/fcitx/fcitx5-bamboo/blob/0f8b3379e95f702cfb2299012d35b388417089d5/bamboo/go.mod#L3

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: Why need this dependency?  > `app-i18n/fcitx-configtool:5`

And, `sys-devel/gettext` is another dependency which is not in the `@system` set, so should be included.

**Comment by bekcpear** [comment]
Please fix the above reviews and rebase all commits to a single commit with a commit message title

```
app-i18n/fcitx-bamboo: new package, add 1.0.1
```

I'll merge it when you complete.

**Review by vuongtuha — COMMENTED** [review]
I finished all code issues.

**Comment by bekcpear** [comment]
> I finished all code issues.

But there is no update in this PR.

**Review by bekcpear — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (vuongtuha) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: I would make it easier to choose language after installing package: just # emerge -av app-i18n/fcitx-bamboo::ryans

Comment (bekcpear) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: This is not the feature of DEPEND in this ebuild, you can use either a meta-package or [a package set](https://wiki.gentoo.org/wiki/Package_sets) to achieve your goal.

The DEPEND of a specific ebuild only serve the building of that package.

Comment (Universebenzene) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: > I would make it easier to choose language after installing package: just # emerge -av app-i18n/fcitx-bamboo::ryans

This is not forcely needed. You can use `optfeature.eclass` to recommend users to install configtools afterwards.

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:14]: You still forget the `-> ${P}.tar.xz`

Should be

```
SRC_URI="https://download.fcitx-im.org/fcitx5/${MY_PN}/${MY_P}.tar.xz -> ${P}.tar.xz"
```

to make the distfile more clear.

**Review by bekcpear — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (vuongtuha) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:14]: I defined MY_P at #9

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: If you write the `pkg_postinst` out and add your own commands, then the default behavior of eclasses in this phase should also be written out. In other words, you should add `xdg_pkg_postinst` in the `pkg_postinst` phase here now, because your own `pkg_postinst` phase overwrites the default one defined in `xdg.eclass` .

**Review by Universebenzene — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: And, `-z` means the following variable is not set, so the following messages will display when this package is updated (disappear when first installed).

Comment (vuongtuha) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: I'm giving up on the suggestion package. Hope the user of this package will be smarter than me T^T

**Review by vuongtuha — COMMENTED** [review]
Hope u review my shortest version :>

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (Universebenzene) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: @bekcpear hmmm, I think when the `REPLACEING_VERSOINS` is true, then all the command after `return` will be ignored. So it seems that he is correct.

Comment (vuongtuha) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: @Universebenzene I run depclean, then emerge new ebuild. It doesn't print out any text. Unless I have a variable active when INSTALL_NEW_PACKAGE (example like that)

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:14]: The MY_P you defined is used to compose the URL, but the filename of the downloaded file is unpredictable. For now is `fcitx5-bamboo-1.0.1.tar.xz` but maybe changed in the feature (depends on upstream maintainer, impossible in most cases but not impossible).

I just want the filename of the distfile stored locally is predictable.

This is just a suggestion, if you insist on not modifying it, it doesn't matter.

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: > @bekcpear hmmm, I think when the `REPLACEING_VERSOINS` is true, then all the command after `return` will be ignored. So it seems that he is correct.

You can test it.

Comment (bekcpear) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: > @Universebenzene I run depclean, then emerge new ebuild. It doesn't print out any text. Unless I have a variable active when INSTALL_NEW_PACKAGE (example like that)

If you want show some notification when first installing it, just replace `-z` with `-n`

**Comment by bekcpear** [comment]
> Hope u review my shortest version :>

After all code reviews finished, please rebase all commits into one, with a suitable message title, e.g.:

```
app-i18n/fcitx-bamboo: new package, add 1.0.1
```

If you are not good at rebase, refer to https://docs.github.com/en/get-started/using-git/using-git-rebase-on-the-command-line

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (Universebenzene) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild:35]: > You can test it.

I was wrong... When I change the `-z ` into `-n`, the postinst phase works as expected... I'm really bad at shell grammar 😭

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: `>`

typo?

**Review by bekcpear — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: You need `xdg_pkg_postinst` before this line.

**Review by Universebenzene — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: missing closing `"`

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: Please use `tab` instead of 8 `space`s

**Review by bekcpear — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: Incorrect License, should be `LGPL-2.1+` 

refer to https://github.com/fcitx/fcitx5-bamboo/tree/main/LICENSES

**Review by bekcpear — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild
Comment (bekcpear) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: And, please also add `MIT`  due to the bamboo-core license: https://github.com/BambooEngine/bamboo-core/blob/master/LICENSE

Comment (vuongtuha) [reply app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: Edited. Please re-check to help me ~~!

**Comment by bekcpear** [comment]
It's better to delete these useless text in the commit message:

```

typo error

update lc

update lc2
```

**Comment by bekcpear** [comment]
Good job!  I'll merge it.

**Pull request merged**
